File: design_patterns/structural/composite/python/src/composite/filesystem/folder.py
from typing import List

from .component import Component


class Folder(Component):
    def __init__(self, name: str) -> None:
        self._name = name
        self._components: List[Component] = []

    def Add(self, component: Component) -> None:
        self._components.append(component)

    def Size(self) -> int:
        total = 0
        for component in self._components:
            total += component.Size()
        return total

File: design_patterns/structural/composite/go/internal/filesystem/folder.go
package filesystem

type Folder struct {
	name       string
	components []Component
}

func (f *Folder) Add(c Component) {
	f.components = append(f.components, c)
}

func (f *Folder) Size() int {
	total := 0
	for _, c := range f.components {
		total += c.Size()
	}
	return total
}

func NewFolder(name string) *Folder {
	return &Folder{
		name: name,
	}
}
